docs/site/Life-cycle.md

```diff
@@ -0,0 +1,212 @@
+---
+lang: en
+title: 'Life cycle events and observers'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Life-cycle.html
+---
+
+## Overview
+
+A LoopBack application has its own life cycles at runtime. There are two methods
+to control the transition of states of `Application`.
+
+- start(): Start the application
+- stop(): Stop the application
+
+It's often desirable for various types of artifacts to participate in the life
+cycles and perform related processing upon `start` and `stop`. Good examples of
+such artifacts are:
+
+- Servers
+
+  - start: Starts the HTTP server listening for connections.
+  - stop: Stops the server from accepting new connections.
+
+- Components
+
+  - A component can register life cycle observers
+
+- DataSources
+
+  - connect: Connect to the underlying database or service
+  - disconnect: Disconnect from the underlying database or service
+
+- Custom scripts
+  - start: Custom logic to be invoked when the application starts
+  - stop: Custom logic to be invoked when the application stops
+
+## The `LifeCycleObserver` interface
+
+To react on life cycle events, a life cycle observer implements the
+`LifeCycleObserver` interface.
+
+```ts
+import {ValueOrPromise} from '@loopback/context';
+
+/**
+ * Observers to handle life cycle start/stop events
+ */
+export interface LifeCycleObserver {
+  preStart?(): ValueOrPromise<void>;
```

Comment:
Wouldn't a dynamic interface (no hard-coded methods) be more flexible? Or do we plan support events only around start and stop?

Some other events that I can think of "ready", "connected", "online" (which was "offline" a while ago), "message"; especially if a component is dependent on other components asynchronously.

Something to think about, since we are considering LB4 as a messaging hub too (can't find your PR).

This makes we wonder if a "global event listener" is what we are looking for.

The interface of `LifeCycleObserver` is "set in stone", but with a "global event listener", we have the flexibility to accommodate an infinite type of events, not just those based around start and stop.